soh/soh/OTRGlobals.cpp

```diff
@@ -1578,24 +1582,56 @@ extern "C" uint32_t OTRGetCurrentHeight() {
-                break;
-            case 2:
-                physicalDevice->SetLedColor(i, {0x64, 0x14, 0x00});
-                break;
-            case 3:
-                physicalDevice->SetLedColor(i, {0x00, 0x3C, 0x64});
-                break;
+        if (i == 0 && physicalDevice->CanSetLed()) {
+            Color_RGB8 color = { 0, 0, 0 };
+            if (brightness > 0.0f) {
+                LEDColorSource source = static_cast<LEDColorSource>(CVarGetInteger("gLedColorSource", LED_SOURCE_TUNIC_ORIGINAL));
+                bool criticalOverride = CVarGetInteger("gLedCriticalOverride", 1);
+                if (criticalOverride || source == LED_SOURCE_HEALTH) {
+                    if (HealthMeter_IsCritical()) {
+                        color = { 0xFF, 0, 0 };
+                    } else if (source == LED_SOURCE_HEALTH) {
+                        if (gSaveContext.health / gSaveContext.healthCapacity <= 0.4f) {
+                            color = { 0xFF, 0xFF, 0 };
+                        } else {
+                            color = { 0, 0xFF, 0 };
+                        }
+                    }
+                }
+                if (gPlayState && (source == LED_SOURCE_TUNIC_ORIGINAL || source == LED_SOURCE_TUNIC_COSMETICS)) {
+                    switch (CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1) {
+                        case PLAYER_TUNIC_KOKIRI:
+                            color = source == LED_SOURCE_TUNIC_COSMETICS
+                                        ? CVarGetColor24("gCosmetics.Link_KokiriTunic.Value", kokiriColor)
+                                        : kokiriColor;
+                            break;
+                        case PLAYER_TUNIC_GORON:
+                            color = source == LED_SOURCE_TUNIC_COSMETICS
+                                        ? CVarGetColor24("gCosmetics.Link_GoronTunic.Value", goronColor)
+                                        : goronColor;
+                            break;
+                        case PLAYER_TUNIC_ZORA:
+                            color = source == LED_SOURCE_TUNIC_COSMETICS
+                                        ? CVarGetColor24("gCosmetics.Link_ZoraTunic.Value", zoraColor)
+                                        : zoraColor;
+                            break;
+                    }
+                }
+                if (source == LED_SOURCE_CUSTOM) {
+                    color = CVarGetColor24("gLedPort1Color", { 255, 255, 255 });
+                }
+                color.r = color.r * brightness;
+                color.g = color.g * brightness;
+                color.b = color.b * brightness;
+            }
+            // We call this every tick, SDL accounts for this use and prevents driver spam
+            // https://github.com/libsdl-org/SDL/blob/f17058b562c8a1090c0c996b42982721ace90903/src/joystick/SDL_joystick.c#L1114-L1144
+            physicalDevice->SetLedColor(i, color);
         }
```

Comment:
```suggestion
    if (physicalDevice->CanSetLed()) {
        auto brightness = CVarGetFloat("gLedBrightness", 1.0f) / 1.0f;
        Color_RGB8 color = { 0, 0, 0 };
        if (brightness > 0.0f) {
            LEDColorSource source = static_cast<LEDColorSource>(CVarGetInteger("gLedColorSource", LED_SOURCE_TUNIC_ORIGINAL));
            bool criticalOverride = CVarGetInteger("gLedCriticalOverride", 1);
            if (criticalOverride || source == LED_SOURCE_HEALTH) {
                if (HealthMeter_IsCritical()) {
                    color = { 0xFF, 0, 0 };
                } else if (source == LED_SOURCE_HEALTH) {
                    if (gSaveContext.health / gSaveContext.healthCapacity <= 0.4f) {
                        color = { 0xFF, 0xFF, 0 };
                    } else {
                        color = { 0, 0xFF, 0 };
                    }
                }
            }
            if (gPlayState && (source == LED_SOURCE_TUNIC_ORIGINAL || source == LED_SOURCE_TUNIC_COSMETICS)) {
                switch (CUR_EQUIP_VALUE(EQUIP_TUNIC) - 1) {
                    case PLAYER_TUNIC_KOKIRI:
                        color = source == LED_SOURCE_TUNIC_COSMETICS
                                    ? CVarGetColor24("gCosmetics.Link_KokiriTunic.Value", kokiriColor)
                                    : kokiriColor;
                        break;
                    case PLAYER_TUNIC_GORON:
                        color = source == LED_SOURCE_TUNIC_COSMETICS
                                    ? CVarGetColor24("gCosmetics.Link_GoronTunic.Value", goronColor)
                                    : goronColor;
                        break;
                    case PLAYER_TUNIC_ZORA:
                        color = source == LED_SOURCE_TUNIC_COSMETICS
                                    ? CVarGetColor24("gCosmetics.Link_ZoraTunic.Value", zoraColor)
                                    : zoraColor;
                        break;
                }
            }
            if (source == LED_SOURCE_CUSTOM) {
                color = CVarGetColor24("gLedPort1Color", { 255, 255, 255 });
            }
            color.r = color.r * brightness;
            color.g = color.g * brightness;
            color.b = color.b * brightness;
        }
        // We call this every tick, SDL accounts for this use and prevents driver spam
        // https://github.com/libsdl-org/SDL/blob/f17058b562c8a1090c0c996b42982721ace90903/src/joystick/SDL_joystick.c#L1114-L1144
        physicalDevice->SetLedColor(0, color);
    }
```

Reply:
It is so hard to tell what's going on sometimes with Github's formatting in the comments... Finally figured it out, though, and I'm left wondering why this change and the previous one were registered as two different changes XD

Reply:
The unfortunate part of that is that it's also tied to the rumble, which isn't limited to the first port only, and leaving it like it is leaves it a little easier to open up to other controller ports should we decide to later on.

Reply:
couple thoughts here:
* why do we set rumble on every controller port? seems kind of odd to rumble all of them when the logic for when we should rumble is only coming from port 1
* similar stuff for the LED logic, if we want to support LED colors on multiple controllers we'd need to rework quite a bit of the logic in there (single brightness cvar, tunic not making sense for controller 2-4 etc.)

I made a PR to your branch (https://github.com/Malkierian/Shipwright/pull/1) with an updated suggestion, hopefully that diff is easier to see